service/src/main/java/greencity/service/FriendServiceImpl.java

```diff
@@ -1,15 +1,49 @@
 package greencity.service;
 
+import greencity.constant.ErrorMessage;
+import greencity.dto.PageableDto;
+import greencity.dto.user.UserFriendDto;
+import greencity.entity.User;
+import greencity.exception.exceptions.NotFoundException;
 import greencity.repository.UserRepo;
-import javax.transaction.Transactional;
+
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Slf4j
 @Service
-@Transactional
+@Transactional(readOnly = true)
 @RequiredArgsConstructor
 public class FriendServiceImpl implements FriendService {
     private final UserRepo userRepo;
+    private final ModelMapper modelMapper;
+
+    public PageableDto<UserFriendDto> findAllUsersFriends(Long userId, Pageable pageable) {
+
+        List<User> friends = userRepo.getAllUserFriends(userId);
```

Comment:
userRepo.getAllUserFriends also uses in the other service so I don`t see a reason for creating a new method with same query

Reply:
You could add a pageable parameter to the query. Because if there is a lot of data passing from db to the server and almost all of it you don't need. With a large amount of data, there will be a lot of unnecessary calculations, etc.